schemas/public/onpremises-kfd-v1alpha2.json

```diff
@@ -1874,10 +1877,87 @@
+        },
+        "backupPrefix": {
+          "type": "string",
+          "additionalProperties": false,
+          "description": "A prefix to be prepended to the backup filenames. If unset, the prefix defaults to the cluster's name."
+        },
+        "persistentVolumeClaim": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": "Configuration for the PersistentVolumeClaim where `etcd-backup-pvc` will store its backups.",
+          "properties": {
+            "schedule": {
+              "type": "string",
+              "description": "The cron expression for the `etcd-backup-pvc` backup schedule. Default is `0 1 * * *`."
+            },
+            "claimName": {
+              "type": "string",
+              "description": "The PersistentVolumeClaim name where the backup will be saved."
+            },
+            "retentionTime": {
+              "type": "string",
+              "description": "The retention time of the backups inside the PersistentVolumeClaim. Follows rclone's `min-age` format. Example: '30d' for 30 days."
+            }
+          }
+        },
+        "s3": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": "Configuration for the remote S3 bucket where `etcd-backup-s3` will send its backups.",
+          "properties": {
+            "schedule": {
+              "type": "string",
+              "description": "The cron expression for the `etcd-backup-s3` backup schedule. Default is `0 1 * * *`."
+            },
+            "endpoint": {
+              "type": "string",
+              "description": "External S3-compatible endpoint for etcd-backup-s3's storage."
+            },
+            "insecure": {
+              "type": "boolean",
+              "description": "If true, will use HTTP as protocol instead of HTTPS."
+            },
+            "secretAccessKey": {
+              "type": "string",
+              "description": "The secret access key (password) for the external S3-compatible bucket."
+            },
+            "accessKeyId": {
+              "type": "string",
+              "description": "The access key ID (username) for the external S3-compatible bucket."
+            },
+            "bucketName": {
+              "type": "string",
+              "description": "The bucket name of the external S3-compatible object storage."
+            },
+            "retentionTime": {
+              "type": "string",
+              "description": "The retention time of the external S3-compatible object storage. Follows rclone's `min-age` format. Example: '30d' for 30 days."
+            }
+          }
+        }
+      }
+    },
```

Comment:
we are missing some conditions here. We should make the schema validate that:

- If etcdBackup.type is pvc, then the PersistenVolumeClaim object must be filled.
- If etcdBackup.type is s3, then the s3 object must be filled.
- If etcdBackup is all, then both PersistentVolumeClaim and s3 must be filled.